src/hera/workflows/script.py

```diff
@@ -48,6 +49,9 @@
 from hera.workflows.steps import Step
 from hera.workflows.task import Task
 
+CLASHING_KWARGS = list(Step.__fields__.keys()) + list(Task.__fields__.keys())
+ARGUMENT_TYPE_KWARGS = ("arguments", "with_param")
```

Comment:
Any chance we can add a small piece of documentation that explains why these 2 fields?

Reply:
Suggesting a small set for deduplication and slightly faster search ```suggestion
CLASHING_KWARGS = set(list(Step.__fields__.keys()) + list(Task.__fields__.keys()))
ARGUMENT_TYPE_KWARGS = ("arguments", "with_param")
```